src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1032,4 +1049,235 @@ private StringConcatFactory() {
         // no instantiation
     }
 
+    /**
+     * Bytecode StringBuilder strategy.
+     *
+     * <p>This strategy emits StringBuilder chains as similar as possible
+     * to what javac would. No exact sizing of parameters or estimates.
+     */
+    private static final class SimpleStringBuilderStrategy {
+        static final int CLASSFILE_VERSION = ClassFile.latestMajorVersion();
```

Comment:
Still breaks backward ASM port, we should use a fixed version like 52 for JAVA_8 and convert to latest only in the CF conversion later.

Reply:
Good catch. In the code I am [reviving](https://github.com/cl4es/jdk/commit/36c4b11bc6cf5a008d5935934aa75f2d2bbe6a23#diff-1339c269a3729d849799d29a7431ccd508a034ced91c1796b952795396843891L771) this field was simply set to `52`. Fixed.